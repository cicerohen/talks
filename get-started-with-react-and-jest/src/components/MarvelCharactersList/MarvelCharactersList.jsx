import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import Loading from 'react-loading';
import List from '../List';
import MarvelCharactersListItem from '../MarvelCharactersListItem';

class MarvelCharactersList extends Component {
  renderCharacter = character => (
    <MarvelCharactersListItem
      key={character.get('id')}
      name={character.get('name')}
      thumbnail={`${character.getIn(['thumbnail', 'path'])}.${character.getIn(['thumbnail', 'extension'])}`}
      description={character.get('description')}
    />
  );
  render() {
    const { characters, isLoading } = this.props;
    return (
      <Fragment>
        {
          isLoading ? (
            <Loading color="#999" type="spin" />
          ) : (
            <List>
              {
                characters
                .map(this.renderCharacter)
                .reduce((acc, curr) => {
                  acc.push(curr);
                  return acc;
                }, [])
              }
            </List>
          )
        }
      </Fragment>
    );
  }
}

MarvelCharactersList.propTypes = {
  characters: PropTypes.objectOf(PropTypes.any).isRequired,
  isLoading: PropTypes.bool.isRequired,
};

export default MarvelCharactersList;
