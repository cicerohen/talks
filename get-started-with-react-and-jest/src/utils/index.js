import { Iterable, fromJS } from 'immutable';
import { createBrowserHistory } from 'history';

export const asImmutable = obj => (Iterable.isIterable(obj) ? obj : fromJS(obj));
export const emptyMap = asImmutable({});

export const getHistory = () => {
    window.browserHistory = window.browserHistory || createBrowserHistory();
    return window.browserHistory;
};

export const isDevEnv = () => {
  const nodeEnv = process.env.NODE_ENV;
  return nodeEnv !== 'production' && nodeEnv !== 'test';
};
