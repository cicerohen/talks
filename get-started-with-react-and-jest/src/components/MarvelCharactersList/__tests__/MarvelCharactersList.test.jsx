import React from 'react';
import { shallow } from 'enzyme';
import { asImmutable } from '../../../utils';
import MarvelCharactersList from '../MarvelCharactersList';

const character = {
  id: 1,
  name: 'character name',
  description: 'character desc',
  thumbnail: 'character thumb',
}

const minimalProps = {
  characters: asImmutable([character]),
  isLoading: false,
}

describe('<MarvelCharactersList /> Tests', () => {
  it('should render without exploding', (done) => {
    const wrapper = shallow(
      <MarvelCharactersList {...minimalProps}/>
    );
    expect(wrapper).toHaveLength(1);
    done();
  });

  it('should render correctly', (done) => {
    const wrapper = shallow(
      <MarvelCharactersList {...minimalProps}/>
    );
    expect(wrapper).toMatchSnapshot();
    done();
  });

  it('should render correctly when is loading', (done) => {
    const wrapper = shallow(
      <MarvelCharactersList {...minimalProps} isLoading={true} />
    );
    expect(wrapper).toMatchSnapshot();
    done();
  });

});
