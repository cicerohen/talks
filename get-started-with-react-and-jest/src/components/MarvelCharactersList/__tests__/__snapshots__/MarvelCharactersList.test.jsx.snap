// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MarvelCharactersList /> Tests should render correctly 1`] = `
<Fragment>
  <List
    style={null}
  >
    <MarvelCharactersListItem
      description="character desc"
      key="1"
      name="character name"
      thumbnail="undefined.undefined"
    />
  </List>
</Fragment>
`;

exports[`<MarvelCharactersList /> Tests should render correctly when is loading 1`] = `
<Fragment>
  <Loading
    color="#999"
    delay={0}
    height={64}
    type="spin"
    width={64}
  />
</Fragment>
`;
